pkg/controller/backup_storage_location_controller.go

```diff
@@ -122,7 +122,7 @@ func (r *BackupStorageLocationReconciler) Reconcile(ctx context.Context, req ctr
 			}
 		}()
 
-		backupStore, err := r.BackupStoreGetter.Get(&location, pluginManager, log)
+		backupStore, err := r.BackupStoreGetter.Get(location.DeepCopy(), pluginManager, log)
```

Comment:
I'd add a comment on why we are doing DeepCopy() here. Alternatively do deep copy in the `objectStoreGetter Get()` function and add a comment there instead. Fixing Get function (so that it actually just get something and not setting anything else) rather than fixing Reconcile function which innocently called Get not expecting change.

```suggestion
		// Using DeepCopy to avoid modifying in-cluster BackupStorageLocation.
		// Get function modifies the location.Spec.Config which we don't want updated to the in-cluster version
		backupStore, err := r.BackupStoreGetter.Get(location.DeepCopy(), pluginManager, log)
```

Reply:
@kaovilai Yeah, I'm leaning towards the latter. The only aspect of the BSL that's passed into the ObjectStore plugin is actually the map of config values. Lets copy *that* map instead of deep copying the whole BSL, and pass it in. No side effect on the kube object. As for 3rd party plugins relying on the BSL in-cluster mod, while I don't think any are doing that now, if they are, it's a dangerous assumption even with the released 1.9 code, since as we've seen with OADP, the change isn't reliably made. Better to ensure it's never made so that breakage is obvious rather than nondeterministic.

Reply:
A "Get" function should never have side effects. If it does have side effects, they had better be well-documented and made with good reason. In this case, they are neither documented nor made with any reason to do so.